templates/repo/header.tmpl

```diff
@@ -32,7 +32,7 @@
 				</div>
 				{{if .CanBeForked}}
 					<div class="ui compact labeled button" tabindex="0">
-						<a class="ui compact button {{if not $.CanSignedUserFork}}poping up{{end}}" {{if $.CanSignedUserFork}}href="{{AppSubUrl}}/repo/fork/{{.ID}}"{{else}} data-content="{{$.i18n.Tr "repo.fork_from_self"}}" data-position="top center" data-variation="tiny"{{end}}>
+						<a class="ui compact button {{if not $.CanSignedUserFork}}poping up{{end}}" {{if $.CanSignedUserFork}}href="{{AppSubUrl}}/repo/fork/{{.ID}}"{{else if $.IsSigned}} data-content="{{$.i18n.Tr "repo.fork_from_self"}}" data-position="top center" data-variation="tiny"{{end}}>
```

Comment:
Wouldn't it be much more clearer if we check IsSigned in the same if as CanSignedUserFork?

Reply:
I'm not understand much but here is the view of a normal user:
It would be great if the behavior is identical to the neighboring buttons "Watch" and "Star".
When I press one of those two buttons I'm asked to sign in.

Reply:
@daviian and not needed IsSigned check, because CanUserFork includes this.

Simplified:
```
<a class="ui compact button {{if not $.CanSignedUserFork}}poping up data-content="{{$.i18n.Tr "repo.fork_from_self"}}" data-position="top center" data-variation="tiny"{{end}}" {{if $.CanSignedUserFork}} href="{{AppSubUrl}}/repo/fork/{{.ID}}"{{end}}>
```

Reply:
@daviian It does not work out(

Reply:
@daviian I keep it in else if because logic I use is:
```
if user can fork & signed in: let them click link
if user can't for & signed in: show them message letting them know not able to fork
if user not signed in: show fork butt with number of forks, but don't let them click
```

Reply:
@flufmonster 
<code>
if user not signed in: let them click and offer the same behavior like the neighboring buttons "Watch" and "Star".
</code>
This would create a uniform operation of all those three buttons at the right upper corner.

Reply:
@oszilloskop sadly we don't know if when the user logs in they can fork, so if we send them to login then it might create a poor experience for them.